textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,664 @@
+	 * fields related to its function. These classes have no methods, they are
+	 * used only as containers to move data.
+	 * Statement --+ SelectStatement
+	 *             + CreateViewStatement
+	 */
+    static public abstract class Statement {
+		public String id;
+      	public Statement() {
+			this(null);
+      	}
+      	public Statement(String id) {
+			this.id = id;
+      	}
+      	@Override
+      	public boolean equals(Object obj) {
+			if (obj == null) { return false; }
+			if (obj.getClass() != this.getClass()) { return false; }
+			Statement kep = (Statement) obj;
+			return new EqualsBuilder()
+						.append(id, kep.id)
+						.isEquals();
+      	}
+    }
+    static public class SelectStatement extends Statement {
+        public Boolean projectAll;
+        public List<String> projectFields;
+        public ExtractPredicate extract;
+        public String from;
+        public Integer limit;
+        public Integer offset;
+      	public SelectStatement() {
+			this(null, null, null, null, null, null, null);
+      	}
+      	public SelectStatement(String id, Boolean projectAll,
+      						List<String> projectFields, ExtractPredicate extract,
+      						String from, Integer limit, Integer offset) {
+			super(id);
+			this.projectAll = projectAll;
+	        this.projectFields = projectFields;
+	        this.extract = extract;
+	        this.from = from;
+	        this.limit = limit;
+	        this.offset = offset;
+      	}
+      	@Override
+      	public boolean equals(Object obj) {
+			if (obj == null) { return false; }
+			if (obj.getClass() != this.getClass()) { return false; }
+			SelectStatement otherSelectStatement = (SelectStatement) obj;
+			return new EqualsBuilder()
+						.appendSuper(super.equals(otherSelectStatement))
+						.append(projectAll, otherSelectStatement.projectAll)
+						.append(projectFields, otherSelectStatement.projectFields)
+						.append(extract, otherSelectStatement.extract)
+						.append(from, otherSelectStatement.from)
+						.append(limit, otherSelectStatement.limit)
+						.append(offset, otherSelectStatement.offset)
+						.isEquals();
+      	}
+    }
+    static public class CreateViewStatement extends Statement {
+        public Statement substatement;
+      	public CreateViewStatement() {
+			this(null, null);
+      	}
+      	public CreateViewStatement(String id, Statement substatement) {
+      	  	super(id);
+	        this.substatement = substatement;
+      	}
+      	@Override
+      	public boolean equals(Object obj) {
+			if (obj == null) { return false; }
+			if (obj.getClass() != this.getClass()) { return false; }
+			CreateViewStatement otherCreateViewStatement = (CreateViewStatement) obj;
+			return new EqualsBuilder()
+						.appendSuper(super.equals(otherCreateViewStatement))
+						.append(substatement, otherCreateViewStatement.substatement)
+						.isEquals();
+      	}
+    }
+
+	/**
+	 * ExtractPredicate class and subclasses(KeywordExtractPredicate)
```

Comment:
Change it to "ExtractPredicate class and its subclasses such as KeywordExtractPredicate"